src/query/queryWithMultiIndex.ts

```diff
@@ -0,0 +1,127 @@
+import { DocumentClient, Key, QueryOutput } from 'aws-sdk/clients/dynamodb';
+import { decrypt, encrypt } from '../utils';
+import { AnyObject, Filter, TableConfig } from '../types';
+import { buildQueryTableParams } from './queryBuilder';
+import { keyBy } from 'lodash';
+
+const DEFAULT_LIMIT = 99999;
+
+export type Queries = {
+  filter: Filter;
+  indexName?: string;
+};
+
+export interface Cursor {
+  [indexName: string]: Key;
+}
+
+export interface Outputs extends QueryOutput {
+  indexName?: string;
+}
+
+/**
+ * Cursor based query which returns list of matching items and the last cursor position
+ *
+ * @param {Queries} queries array of query filter with index
+ * @param { QueryWithCursorOptions } options cursor, page size options
+ * @param {number} limit number of items
+ * @param {string} prevCursor starting cursor position
+ * @returns {Array<T>, string} list of matching items, last cursor position
+ */
+export async function queryWithMultiIndex<T extends AnyObject>(
```

Comment:
This function is big,
Can you make it cleaner?
 
Make smaller functions and give good names.